### PR TITLE
Bump mysqlclient to support the 1.4.x and 2.x series

### LINKED ISSUE #8292: Bump supported mysqlclient to <1.5
**Description**

version 1.4.X introduced in Jan 2019 

we should support it if we can.

**Use case / motivation**

pin of <1.4 was done in https://github.com/apache/airflow/pull/4558 due to lack of Python 2 compatibility. Since Master doesn't support Python 2 anymore there is no need for that restriction

**Related Issues**

Moved from https://issues.apache.org/jira/browse/AIRFLOW-4810

I tried to fix it in https://github.com/apache/airflow/pull/5430 but didn't get help with the tests so if any one wants to pick it up be my guest.

### PR DESCRIPTION
**Pull request by uranusjr**
Another attempt to resolve #8292.

### CODE REVIEW

**Comment by kaxil** [comment]
Please rebase on latest Master, failing doc build should be fixed by https://github.com/apache/airflow/pull/14986

**Comment by kaxil** [comment]
Just rebased on latest Master, should fix failing issues now 🤞

**Comment by ashb** [comment]
@uranusjr May be worth updating the dep to `<3` now actually. The only breaking change form 1.x to 2.x was removal of of py2 support https://github.com/PyMySQL/mysqlclient/blob/master/HISTORY.rst

**Comment by uranusjr** [comment]
Updated. I should have checked the version compatibility first!

**Review by ashb — APPROVED** [review]

File: docs/apache-airflow-providers-mysql/index.rst [hunk 1]
```diff
@@ -93,7 +93,7 @@ PIP requirements
 PIP package                 Version required
 ==========================  ======================
 ``mysql-connector-python``  ``>=8.0.11, <=8.0.22``
-``mysqlclient``             ``>=1.3.6,<1.4``
+``mysqlclient``             ``>=1.3.6,<3``
```
Comment (eladkal) [inline docs/apache-airflow-providers-mysql/index.rst:96]: I think the package versions at this doc are auto generated when provider is released and shouldn't be update manually. @potiuk is that correct?

Comment (ashb) [reply docs/apache-airflow-providers-mysql/index.rst:96]: I think they are, but I also don't think there's any harm in editing it as part of this PR.

**Comment by ashb** [comment]
A bug in our CI might have meant that these tests were run against the _old_ dep, and as a result one of the GCS tests fails because it tried to do:

```
from _mysql_exceptions import ProgrammingError
```

**Pull request merged**